ext/gd/gd.c

```diff
@@ -4138,15 +4138,33 @@ PHP_FUNCTION(imageaffinematrixget)
 				RETURN_THROWS();
 			}
 
-			if ((tmp = zend_hash_str_find(Z_ARRVAL_P(options), "x", sizeof("x") - 1)) != NULL) {
-				x = zval_get_double(tmp);
+			if ((tmp = zend_hash_str_find_deref(Z_ARRVAL_P(options), "x", sizeof("x") - 1)) != NULL) {
+				switch (Z_TYPE_P(tmp)) {
+				case IS_LONG:
+				case IS_DOUBLE:
+				case IS_STRING:
+					x = zval_get_double(tmp);
+					break;
```

Comment:
Won't this return `0` for garbage strings like `"string"` ? Obviously an improvement over the current situation, but still not amazing. Also this probably will fail for GMP objects which might be supported rn?